datahub-web-react/src/app/lineage/manage/ManageLineageMenu.tsx

```diff
@@ -0,0 +1,173 @@
+}
+
+interface Props {
+    entityUrn: string;
+    refetchEntity: () => void;
+    setUpdatedLineages: React.Dispatch<React.SetStateAction<UpdatedLineages>>;
+    disableUpstream?: boolean;
+    disableDownstream?: boolean;
+    centerEntity?: () => void;
+    showLoading?: boolean;
+    menuIcon?: React.ReactNode;
+    entityType?: EntityType;
+    entityPlatform?: string;
+    canEditLineage?: boolean;
+    disableDropdown?: boolean;
+}
+
+export default function ManageLineageMenu({
+    entityUrn,
+    refetchEntity,
+    setUpdatedLineages,
+    disableUpstream,
+    disableDownstream,
+    centerEntity,
+    showLoading,
+    menuIcon,
+    entityType,
+    entityPlatform,
+    canEditLineage,
+    disableDropdown,
+}: Props) {
+    const [isModalVisible, setIsModalVisible] = useState(false);
+    const [lineageDirection, setLineageDirection] = useState<Direction>(Direction.Upstream);
+
+    function manageLineage(direction: Direction) {
+        setIsModalVisible(true);
+        setLineageDirection(direction);
+    }
+
+    const isCenterNode = !disableUpstream && !disableDownstream;
+    const isDashboard = entityType === EntityType.Dashboard;
+    const isDownstreamDisabled = disableDownstream || isDashboard || !canEditLineage;
+    const isUpstreamDisabled = disableUpstream || !canEditLineage;
+    const isManualLineageSupported = entityType && ENTITY_TYPES_WITH_MANUAL_LINEAGE.has(entityType);
+
+    // if we don't show manual lineage options or the center node option, this menu has no options
+    if (!isManualLineageSupported && isCenterNode) return null;
+
+    return (
+        <>
+            <Tooltip title={disableDropdown ? UNAUTHORIZED_TEXT : ''}>
+                <div>
+                    <Dropdown
+                        overlayStyle={{ zIndex: DROPDOWN_Z_INDEX }}
+                        disabled={disableDropdown}
+                        overlay={
+                            <Menu>
+                                {isManualLineageSupported && (
+                                    <>
+                                        <Popover
+                                            content={
+                                                !canEditLineage ? (
+                                                    UNAUTHORIZED_TEXT
+                                                ) : (
+                                                    <PopoverContent centerEntity={centerEntity} direction="upstream" />
+                                                )
+                                            }
+                                            overlayStyle={
+                                                isUpstreamDisabled ? { zIndex: POPOVER_Z_INDEX } : { display: 'none' }
+                                            }
+                                        >
+                                            <div>
+                                                <Menu.Item
```

Comment:
minor: It would have been nice to factor some of these items out. maybe in a follow up

Reply:
agreed - it got a good amount busier at the last leg of this feature with adding more and more to the menu - a little cleanup wouldn't hurt anybody